src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -934,17 +971,21 @@ def get_lock(name, resource_group_name=None, resource_provider_namespace=None,
                                      resource_type, resource_name)
 
     if resource_group_name is None:
-        return lock_client.management_locks.get_at_subscription_level(name)
+        return _call_correct_subscription_get(lock_client, name)
     if resource_name is None:
         return lock_client.management_locks.get_at_resource_group_level(resource_group_name, name)
+    if supported_api_version(ResourceType.MGMT_RESOURCE_LOCKS, max_api='2015-01-01'):
+        lock_list = list_locks(resource_group_name, resource_provider_namespace, parent_resource_path,
+                               resource_type, resource_name)
+        lock_list = [lock for lock in lock_list if lock.name == name]
+        return lock_list[0] if lock_list else None
```

Comment:
Following form is more concise and performant. Replacing above two lines.
```
return next((lock for lock in lock_list if lock.name == name), None)
```
Reference: https://docs.python.org/2/library/functions.html#next